client/rest-high-level/src/test/java/org/elasticsearch/client/IndexLifecycleIT.java

```diff
@@ -169,35 +169,30 @@ public void testStartStopILM() throws Exception {
         createIndex("baz", Settings.builder().put("index.lifecycle.name", "eggplant").build());
         createIndex("squash", Settings.EMPTY);
 
-        // TODO: NORELEASE convert this to using the high level client once
-        // there are APIs for it
-        Request statusReq = new Request("GET", "/_ilm/status");
-        Response statusResponse = client().performRequest(statusReq);
-        String statusResponseString = EntityUtils.toString(statusResponse.getEntity());
-        assertEquals("{\"operation_mode\":\"RUNNING\"}", statusResponseString);
+        TimedRequest statusRequest = new TimedRequest();
```

Comment:
While I get the logic behind why you did it this way, I think generic classes make more sense for responses than requests.  As a user of the HLRC, this feels really strange - even if `GetILMStatusRequest` would be an extra class that doesn't really add anything, I think having separate classes for each request type is the way to go.  I could be wrong though, I'd like to hear others' input on this.

Reply:
I don't have a strong opinion beyond consistency in the API.  

Perhaps we should allow a no-arg method in the client that creates the request for them. However, that also creates inconsistency (unless you create a bunch of new methods).

> I'd like to hear others' input on this.

..same

Reply:
@hub-cap Could you weigh in on this?